gems/sorbet-runtime/lib/types/types/typed_enumerable.rb

```diff
@@ -50,7 +50,7 @@ def valid?(obj)
         # Enumerators can be unbounded: see `[:foo, :bar].cycle`
         return true
       when Range
-        @type.valid?(obj.first) && @type.valid?(obj.last)
+        @type.valid?(obj.begin) && @type.valid?(obj.end)
```

Comment:
I don't think this is going to work, because if the range is endless, `obj.end` will be `nil` but `nil` will not be a valid `Integer` for e.g. `T::Range[Integer]`

Reply:
Can you try adding a test like the one you wrote in #3093?

You can look in `method_validation.rb` for how to write tests that test calling real methods with real sigs by giving them real values (rather than by calling into the internals)

Reply:
Right -- I'll go with the implementation approach you suggested in the original issue 👌